src/com/boggle/Main.java

```diff
@@ -6,8 +6,267 @@
+
+                    if (i != row && a != col){
+                        path += wordArray[i][a];
+                        answer = checkString(path);
+                        if (answer = true){
+                            System.out.println(path);
+                            explore(i,a, path);
+                        }
+                    }
+                }
+            }
+        }
+
+        else {
+            path += wordArray[row][col];
+            if (pos == "side Upper"){
+
+                for (int i = row; i <= row + 1; i++){
+                    for (int a = col - 1; a <= col+1; a++){
+                        if (i!= row && a!= col){
+                            path += wordArray[i][a];
+                            answer = checkString(path);
+                            if (answer == true){
+                                System.out.println(path);
+                                explore(i,a, path);
+                            }
+                        }
+                    }
+                }
+
+
+            }
+
+            else if (pos == "side Right"){
+
+                for (int i = row - 1; i <= row + 1; i++){
+                    for (int a = col - 1; a <= col; a++){
+                        if (i!= row && a!= col){
+                            path += wordArray[i][a];
+                            answer = checkString(path);
+                            if (answer == true){
+                                System.out.println(path);
+                                explore(i,a, path);
+                            }
+                        }
+                    }
+                }
+
+
+
+
+            }
+            else if (pos == "side Lower"){
+
+                for (int i = row - 1; i <= row; i++){
+                    for (int a = col - 1; a <= col+1; a++){
+                        if (i!= row && a!= col){
+                            path += wordArray[i][a];
+                            answer = checkString(path);
+                            if (answer == true){
+                                System.out.println(path);
+                                explore(i,a, path);
+                            }
+                        }
+                    }
+                }
+
+
+            }
+
+            else if (pos == "side Left"){
+
+                for (int i = row - 1; i <= row + 1; i++){
+                    for (int a = col; a <= col+1; a++){
+                        if (i!= row && a!= col){
+                            path += wordArray[i][a];
+                            answer = checkString(path);
+                            if (answer == true){
+                                System.out.println(path);
+                                explore(i,a, path);
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        return path;
     }
 
     public static Boolean checkString(String word) {
```

Comment:
i think the two helper functions you need here are: `checkIsWord` and `checkIsWordPrefix`, can you explain why youd need both and how they are different?

Reply:
checkIsWord will be checking if the letters have fully spelled out one word, whereas checkis WordPrefix will check if the combination of letters form the beginning of a word.